pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go

```diff
@@ -116,3 +137,156 @@ func TestIsRequiredConfigPresent(t *testing.T) {
+				t.Errorf("created Kube API Pod is different from the expected one (form a golden file) : %s", diff.ObjectDiff(actualKASPod, goldenKASPod))
+			}
+		})
+	}
+}
+
+func TestManageKubeAPIServerConfig(t *testing.T) {
+	var ocpScheme = runtime.NewScheme()
+	var ocpCodec = serializer.NewCodecFactory(ocpScheme)
+	if err := kubecontrolplanev1.Install(ocpScheme); err != nil {
+		t.Fatal(err)
+	}
+
+	scenarios := []struct {
+		name         string
+		goldenFile   string
+		operatorSpec *operatorv1.StaticPodOperatorSpec
+		platformType configv1.PlatformType
+	}{
+
+		// scenario 1
+		{
+			name:         "happy path: a default config map is created",
+			goldenFile:   "./testdata/kas_config_scenario_1.yaml",
+			operatorSpec: &operatorv1.StaticPodOperatorSpec{OperatorSpec: operatorv1.OperatorSpec{}},
+		},
+
+		// scenario 2
+		{
+			name:         "the shutdown-delay-duration is extended due to a known AWS issue: https://bugzilla.redhat.com/show_bug.cgi?id=1943804a",
+			goldenFile:   "./testdata/kas_config_scenario_2.yaml",
+			operatorSpec: &operatorv1.StaticPodOperatorSpec{OperatorSpec: operatorv1.OperatorSpec{}},
+			platformType: configv1.AWSPlatformType,
+		},
+	}
+
+	for _, scenario := range scenarios {
+		t.Run(scenario.name, func(t *testing.T) {
+			// test data
+			infrastructureIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+			infrastructureIndexer.Add(&configv1.Infrastructure{ObjectMeta: metav1.ObjectMeta{Name: "cluster"}, Spec: configv1.InfrastructureSpec{PlatformSpec: configv1.PlatformSpec{Type: scenario.platformType}}})
+			infrastructureLister := configlistersv1.NewInfrastructureLister(infrastructureIndexer)
+
+			// act
+			actualConfigMap, err := manageKubeAPIServerConfigInternal(scenario.operatorSpec, infrastructureLister)
+
+			// validate
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			serializedKASConfig, exist := actualConfigMap.Data["config.yaml"]
+			if !exist {
+				t.Fatal("didn't find config.yaml")
+			}
+			decodedKASConfig, err := runtime.Decode(ocpCodec.UniversalDecoder(kubecontrolplanev1.GroupVersion, configv1.GroupVersion, osinv1.GroupVersion), []byte(serializedKASConfig))
+			if err != nil {
+				t.Fatal(err)
+			}
+			actualKASConfig := decodedKASConfig.(*kubecontrolplanev1.KubeAPIServerConfig)
+
+			data := readBytesFromFile(t, scenario.goldenFile)
+			goldenKASConfig := &kubecontrolplanev1.KubeAPIServerConfig{}
+			if err := runtime.DecodeInto(codec, data, goldenKASConfig); err != nil {
+				t.Fatal(err)
+			}
+
+			if !equality.Semantic.DeepEqual(actualKASConfig, goldenKASConfig) {
+				t.Errorf("created Kube API Config is different from the expected one (form a golden file) : %s", diff.ObjectDiff(actualKASConfig, goldenKASConfig))
+			}
+		})
+	}
+}
+
+func readBytesFromFile(t *testing.T, filename string) []byte {
+	file, err := os.Open(filename)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer file.Close()
+
+	data, err := ioutil.ReadAll(file)
```

Comment:
TIOLI: `ioutil.ReadFile(filename)` (or `os.ReadFile()` for Go 1.15+) can make this simpler.

Reply:
done